### PR TITLE
feat: [CDS-46005]: support runtime variables in Merge Step

### PR DESCRIPTION
**Pull request by ishantmehtaharness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by ishantmehtaharness** [comment]
dockerize

File: src/modules/75-cd/components/PipelineSteps/MergePrStep/MergePrInputStep.tsx [hunk 1]
```diff
@@ -59,6 +68,59 @@ export default function MergePRInputStep(props: MergePrInputStepProps): React.Re
...
+            defaultValueToReset={[]}
+            disableTypeSelection
+          >
+            <FieldArray
+              name="spec.variables"
+              render={() => {
+                return (
+                  <div className={css.panel}>
+                    <div className={css.environmentVarHeader}>
+                      <span className={css.label}>Name</span>
```
Comment (cjlee01) [inline src/modules/75-cd/components/PipelineSteps/MergePrStep/MergePrInputStep.tsx:85]: NIT: should these be in the strings.en.yaml file

**Review by cjlee01 — DISMISSED** [review]

File: src/modules/75-cd/components/PipelineSteps/MergePrStep/MergePrInputStep.tsx [hunk 2]
```diff
@@ -59,6 +68,59 @@ export default function MergePRInputStep(props: MergePrInputStepProps): React.Re
           />
         </div>
       )}
+      {isArray(get(template, 'spec.variables', [])) && get(template, 'spec.variables', []).length ? (
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/PipelineSteps/MergePrStep/MergePrInputStep.tsx:71]: we can extract it out in a common variable for better readability

const variables = get(template, 'spec.variables', [])

and then we can check if 'variables' is a non empty array

File: src/modules/75-cd/components/PipelineSteps/MergePrStep/MergePrInputStep.tsx [hunk 3]
```diff
@@ -59,6 +68,59 @@ export default function MergePRInputStep(props: MergePrInputStepProps): React.Re
...
+              name="spec.variables"
+              render={() => {
+                return (
+                  <div className={css.panel}>
+                    <div className={css.environmentVarHeader}>
+                      <span className={css.label}>Name</span>
+                      <span className={css.label}>Type</span>
+                      <span className={css.label}>Value</span>
+                    </div>
+                    {template?.spec?.variables?.map((type, i: number) => {
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/PipelineSteps/MergePrStep/MergePrInputStep.tsx:89]: https://github.com/harness/harness-core-ui/pull/12396/files#r1020517345
 
we can use the same var here as well

That will also help us to remove the redundant optional chaining - as at this point we already know that 'variables' is a non empty array

Comment (ishantmehtaharness) [reply src/modules/75-cd/components/PipelineSteps/MergePrStep/MergePrInputStep.tsx:89]: Good point @samarth-sharma31 , we know its an array already.
I have incorporated the changes. Thanks for the review man !!

**Review by cjlee01 — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/MergePrStep/MergePrInputStep.tsx (hunk 2 above)
Comment (ishantmehtaharness) [reply src/modules/75-cd/components/PipelineSteps/MergePrStep/MergePrInputStep.tsx]: resolved

**Review by samarth-sharma31 — APPROVED** [review]

**Pull request merged**